lib/drivers/medtronic600Driver.js

```diff
@@ -1332,8 +1716,91 @@ module.exports = (config) => {
     processData(progress, data, cb) {
       debug('in processData');
       cfg.builder.setDefaults({
-        deviceId: driver.pumpSession.bcnlModelAndSerial,
+        deviceId: `${data.settings.pumpModel}:${data.settings.pumpSerial}`,
       });
+
+      // TODO - add more time change records
+      const currentDeviceTime = data.settings.currentDeviceTime;
+      cfg.tzoUtil = new TZOUtil(cfg.timezone, currentDeviceTime, []);
+
+      data.postRecords = [];
+      const postSettings = cfg.builder.makePumpSettings();
+      postSettings.with_units(data.settings.units)
+        .with_carbRatio(data.settings.carbRatio)
+        .with_insulinSensitivity(data.settings.insulinSensitivity)
+        .with_bgTarget(data.settings.bgTarget)
+        .with_basalSchedules(data.settings.basalSchedules)
+        .with_activeSchedule(data.settings.activeSchedule)
+        .with_time(currentDeviceTime)
+        .with_deviceTime(sundial.formatDeviceTime(currentDeviceTime))
+        .with_timezoneOffset(sundial.getOffsetFromZone(currentDeviceTime, cfg.timezone))
+        .with_conversionOffset(0);
+      data.postRecords.push(postSettings.done());
+
+      let events = [];
+      new NGPHistoryParser(cfg, data.pages.concat(data.cbg_pages))
```

Comment:
Hmm, maybe I should rename my `processData.js` in the 5/7-series as `HistoryParser` - more descriptive.